Mathlib/Analysis/Asymptotics/TVS.lean

```diff
@@ -113,101 +160,273 @@ theorem IsLittleOTVS.congr (h : f₁ =o[𝕜;l] g₁) (hf : ∀ x, f₁ x = f₂
-    (hV₂ ε hε).mono fun x hx ↦ hx.trans <| by gcongr
+  (isLittleOTVS_iff ..).trans <| forall₂_congr fun U hU ↦ .symm <|
+    eventually_smallSets' fun V₁ V₂ hV hV₂ ε hε ↦ (hV₂ ε hε).mono fun x hx ↦ hx.trans <| by gcongr
 
 alias ⟨IsLittleOTVS.eventually_smallSets, _⟩ := isLittleOTVS_iff_smallSets
 
+theorem isBigOTVS_iff_smallSets :
+    f =O[𝕜; l] g ↔ ∀ U ∈ 𝓝 0, ∀ᶠ V in (𝓝 0).smallSets,
+      ∀ᶠ x in l, egauge 𝕜 U (f x) ≤ egauge 𝕜 V (g x) :=
+  (isBigOTVS_iff ..).trans <| forall₂_congr fun U hU ↦ .symm <|
+    eventually_smallSets' fun V₁ V₂ hV hV₂ ↦ hV₂.mono fun x hx ↦ hx.trans <| by gcongr
+
+alias ⟨IsBigOTVS.eventually_smallSets, _⟩ := isBigOTVS_iff_smallSets
+
 @[simp]
 theorem isLittleOTVS_map {k : β → α} {l : Filter β} :
     f =o[𝕜; map k l] g ↔ (f ∘ k) =o[𝕜;l] (g ∘ k) := by
-  simp [IsLittleOTVS]
+  simp [isLittleOTVS_iff, EventuallyLE]
+
+@[simp]
+theorem isBigOTVS_map {k : β → α} {l : Filter β} :
+    f =O[𝕜; map k l] g ↔ (f ∘ k) =O[𝕜;l] (g ∘ k) := by
+  simp [isBigOTVS_iff, EventuallyLE]
 
 lemma IsLittleOTVS.mono (hf : f =o[𝕜;l₁] g) (h : l₂ ≤ l₁) : f =o[𝕜;l₂] g :=
-  fun U hU => let ⟨V, hV0, hV⟩ := hf U hU; ⟨V, hV0, fun ε hε => (hV ε hε).filter_mono h⟩
+  ⟨fun U hU ↦ let ⟨V, hV0, hV⟩ := hf.1 U hU; ⟨V, hV0, fun ε hε ↦ (hV ε hε).filter_mono h⟩⟩
+
+lemma IsBigOTVS.mono (hf : f =O[𝕜;l₁] g) (h : l₂ ≤ l₁) : f =O[𝕜;l₂] g :=
+  ⟨fun U hU ↦ let ⟨V, hV0, hV⟩ := hf.1 U hU; ⟨V, hV0, hV.filter_mono h⟩⟩
 
 lemma IsLittleOTVS.comp_tendsto {k : β → α} {lb : Filter β} (h : f =o[𝕜; l] g)
     (hk : Tendsto k lb l) : (f ∘ k) =o[𝕜; lb] (g ∘ k) :=
   isLittleOTVS_map.mp (h.mono hk)
 
+lemma IsBigOTVS.comp_tendsto {k : β → α} {lb : Filter β} (h : f =O[𝕜; l] g)
+    (hk : Tendsto k lb l) : (f ∘ k) =O[𝕜; lb] (g ∘ k) :=
+  isBigOTVS_map.mp (h.mono hk)
+
 lemma isLittleOTVS_sup : f =o[𝕜; l₁ ⊔ l₂] g ↔ f =o[𝕜; l₁] g ∧ f =o[𝕜; l₂] g := by
   simp only [isLittleOTVS_iff_smallSets, ← forall_and, ← eventually_and, eventually_sup]
 
 lemma IsLittleOTVS.sup (hf₁ : f =o[𝕜; l₁] g) (hf₂ : f =o[𝕜; l₂] g) : f =o[𝕜; l₁ ⊔ l₂] g :=
   isLittleOTVS_sup.mpr ⟨hf₁, hf₂⟩
 
+lemma _root_.ContinuousLinearMap.isBigOTVS_id {l : Filter E} (f : E →L[𝕜] F) : f =O[𝕜; l] id :=
+  ⟨fun U hU ↦ ⟨f ⁻¹' U, (map_continuous f).tendsto' 0 0 (map_zero f) hU, .of_forall <|
+    (mapsTo_preimage f U).egauge_le 𝕜 f⟩⟩
+
+lemma _root_.ContinuousLinearMap.isBigOTVS_comp (g : E →L[𝕜] F) : (g <| f ·) =O[𝕜; l] f :=
+  g.isBigOTVS_id.comp_tendsto tendsto_top
+
 @[simp]
 lemma IsLittleOTVS.zero (g : α → F) (l : Filter α) : (0 : α → E) =o[𝕜;l] g := by
-  intros U hU
-  simpa [egauge_zero_right _ (Filter.nonempty_of_mem hU)] using ⟨univ, by simp⟩
+  refine ⟨fun U hU ↦ ?_⟩
+  use univ
+  simp [egauge_zero_right _ (Filter.nonempty_of_mem hU), EventuallyLE]
 
 lemma isLittleOTVS_insert [TopologicalSpace α] {x : α} {s : Set α} (h : f x = 0) :
-    f =o[𝕜;(𝓝[insert x s] x)] g ↔ f =o[𝕜;(𝓝[s] x)] g := by
+    f =o[𝕜; 𝓝[insert x s] x] g ↔ f =o[𝕜;(𝓝[s] x)] g := by
```

Comment:
I don't think I agree with adding these spaces, as they're not present in the infoview

Reply:
Unless you've decided the notation should _always_ use a space, in which case you should probably make a standalone PR that makes that change to the `notation` and cleans up all the uses of it.

Reply:
OK, I'll open another PR for this. I think that we should put spaces after punctuation.

Reply:
Opened #24425

Reply:
I've merged `origin/staging` and pushed before I remembered that it triggers a bug in github browser. I'll push some 1-line change once #24425 is merged to make github compare to `master`.

Reply:
Done